### ISSUE TITLE
Split impl.py

### ISSUE THREAD

**Issue by Jrmyy** [dbt-athena/dbt-athena#443]
As the `impl.py` is getting bigger and bigger (more than 800 lines right now), we should find a way to reorganise our python files. Should it be by aws service (s3, glue, lf, athena) ? Should we have a package dbt for dbt-related overrides and an aws package for interaction with boto, pyathena libraries (and then split up by file) and a third package for utils ?

**Comment by nicor88** [comment]
I believe that splitting by AWS service and other separate files is fine. I won't go for the path of using other libraries, as it will make the friction to contribute higher and the setup harder to maintain.

**Comment by Jrmyy** [comment]
Here is what I had in mind:
```bash
.
├── aws
│   ├── __init__.py
│   ├── config.py
│   ├── session.py
│   ├── utils.py
│   ├── athena
│   │   ├── __init__.py
│   │   └── ...
│   ├── glue
│   │   └── __init__.py
│   │   └── ...
│   ├── lakeformation
│   │   └── __init__.py
│   │   └── ...
│   ├── s3
│   │   └── __init__.py
│   │   └── ...
│   └── sts
│       └── __init__.py
│       └── ...
├── dbt
│   ├── __init__.py
│   ├── adapter_response.py
│   ├── column.py
│   ├── connection_manager.py
│   ├── credentials.py
│   ├── formatter.py
│   ├── query_headers.py
│   └── relation.py
│   └── ...
├── constants.py
└── exceptions.py
```

**Comment by Jrmyy** [comment]
We can try without subfolders, I just want to be sure we don't have files with too many lines but maybe it is too much overhead for the moment one package per aws service

**Comment by nicor88** [comment]
> We can try without subfolders, I just want to be sure we don't have files with too many lines but maybe it is too much overhead for the moment one package per aws service
fair enough - we can consider to use subfolders for services where one file is too much, a hybrid approach then.
